SerialPrograms/Source/Integrations/DppIntegration/DppClient.cpp

```diff
@@ -118,14 +121,15 @@ void Client::send_embed_dpp(
 }
 
 void Client::run(const std::string& token) {
+    std::lock_guard<std::mutex> lg(m_client_lock);
```

Comment:
After merging this, I'll add a null check for `m_bot` here. It can be null if `disconnect()` gets called immediately after `connect()`, but before the thread can start up.

`connect()` - Constructs `m_bot` and creates the thread.
`disconnect()` - Destructs `m_bot` and sets it back to null.
`run()` - Thread finally starts running and sees that `m_bot` is null.